src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -4256,7 +4261,66 @@ private void GenerateOneCode()
                     //:         break Backward;
                     {
                         string str = _strings![Operand(0)];
+                        Label multiCode = DefineLabel();
+                        if (!IsRightToLeft())
+                        {
+                            // if (runtextend - runtextpos < c)
+                            Ldloc(_runtextendLocal!);
+                            Ldloc(_runtextposLocal!);
+                            Sub();
+                            Ldc(str.Length);
+                            BgeFar(multiCode);
+                            // if (!_caseInsensitive && _maxBacktrackPosition != -1 && runtextpos > _maxBacktrackPosition)
+                            if (!IsCaseInsensitive())
+                            {
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Ldc(-1);
+                                BeqFar(_backtrack);
+                                Ldloc(_runtextposLocal!);
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                BleFar(_backtrack);
+                                // runtextpos = _maxBacktrackPosition;
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Stloc(_runtextposLocal!);
+                                // ReadOnlySpan<char> runtextSpan = runtext.AsSpan(_maxBacktrackPosition, runtextend - _maxBacktractPosition);
+                                Ldloc(_runtextLocal!);
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Ldloc(_runtextendLocal!);
+                                Ldloc(_maxBacktrackPositionLocal!);
+                                Sub();
+                                using (RentedLocalBuilder textSpanLocal = RentReadOnlySpanCharLocal())
+                                {
+                                    Call(s_stringAsSpanIntIntMethod);
+                                    Stloc(textSpanLocal);
+                                    using (RentedLocalBuilder lastIndexOf = RentInt32Local())
+                                    {
+                                        using (RentedLocalBuilder strAsSpanLocal = RentReadOnlySpanCharLocal())
+                                        {
+                                            // int lastIndexOf = runtextSpan.LastIndexOf(str.AsSpan());
+                                            Ldstr(str);
+                                            Call(s_stringAsSpanMethod);
+                                            Stloc(strAsSpanLocal);
+                                            Ldloc(textSpanLocal);
+                                            Ldloc(strAsSpanLocal);
+                                            Call(s_spanLastIndexOfMethod);
+                                        }
```

Comment:
Why do we need this strAsSpanLocal?  Couldn't this instead be:
```C#
// int lastIndexOf = runtextSpan.LastIndexOf(str.AsSpan());
Ldloc(textSpanLocal);
Ldstr(str);
Call(s_stringAsSpanMethod);
Call(s_spanLastIndexOfMethod);
```
?

Reply:
It could. Fixed!